dhis-2/dhis-test-integration/src/test/java/org/hisp/dhis/dataapproval/DataApprovalWorkflowServiceTest.java

```diff
@@ -172,7 +171,7 @@ void testSaveWorkFlowWithAuthority() {
   void testSaveWorkFlowWithoutAuthority() {
     createUserAndInjectSecurityContext(false, null);
     assertThrows(
-        CreateAccessDeniedException.class,
+        AccessDeniedException.class,
```

Comment:
Note that `assertThrows` does include sub-classes so if this still would throw a `CreateAccessDeniedException` this assert would also pass. If you want to assert an exact exception type use `assertThrowsExactly`. I always thought the naming on those got it wrong because I feel the default one with the short name should have been the exact match and then there should be one called `assertThrowsAssignable` or so if you want to include any subtypes.

Reply:
`CreateAccessDeniedException` will never be thrown as it has been deleted. It's now either `ForbiddenException` or `AccessDeniedException` depending on the level where it's thrown.